testbed/tests/app/test_desktop.py

```diff
@@ -168,159 +170,217 @@ async def test_menu_minimize(app, app_probe):
-    assert app_probe.content_size(window1)[0] > 1000
-    assert app_probe.content_size(window1)[1] > 700
-    assert (
-        window1_widget_probe.width > initial_window1_widget_size[0]
-        and window1_widget_probe.height > initial_window1_widget_size[1]
-    )
-
-    assert not app_probe.is_full_screen(window2)
-    assert app_probe.content_size(window2) == initial_content2_size
-    assert (
-        window2_widget_probe.width == initial_window2_widget_size[0]
-        and window2_widget_probe.height == initial_window2_widget_size[1]
-    )
-
-    # Exit full screen
-    app.exit_full_screen()
-    await window1_probe.wait_for_window(
-        "No longer full screen",
-        full_screen=True,
-    )
-
-    assert not app.is_full_screen
-
-    assert not app_probe.is_full_screen(window1)
-    assert app_probe.content_size(window1) == initial_content1_size
-    assert (
-        window1_widget_probe.width == initial_window1_widget_size[0]
-        and window1_widget_probe.height == initial_window1_widget_size[1]
-    )
-
-    assert not app_probe.is_full_screen(window2)
-    assert app_probe.content_size(window2) == initial_content2_size
-    assert (
-        window2_widget_probe.width == initial_window2_widget_size[0]
-        and window2_widget_probe.height == initial_window2_widget_size[1]
-    )
-
-    # Go full screen again on window 1
-    app.set_full_screen(window1)
-    # A longer delay to allow for genie animations
-    await window1_probe.wait_for_window(
-        "First extra window is full screen",
-        full_screen=True,
-    )
-    assert app.is_full_screen
-
-    assert app_probe.is_full_screen(window1)
-    assert app_probe.content_size(window1)[0] > 1000
-    assert app_probe.content_size(window1)[1] > 700
-    assert (
-        window1_widget_probe.width > initial_window1_widget_size[0]
-        and window1_widget_probe.height > initial_window1_widget_size[1]
-    )
+async def test_presentation_mode(app, app_probe, main_window, main_window_probe):
+    """The app can enter presentation mode."""
+    try:
+        window_information_list = list()
+        windows_list = list()
+        for i in range(len(app.screens)):
+            window = toga.Window(
+                title=f"Test Window {i}",
+                position=(150 + (10 * i), 150 + (10 * i)),
+                size=(200, 200),
+            )
+            r = random.randint(0, 255)
+            g = random.randint(0, 255)
+            b = random.randint(0, 255)
+            window_widget = toga.Box(style=Pack(flex=1))
+            window.content = toga.Box(
+                children=[window_widget],
+                style=Pack(background_color=f"#{r:02X}{g:02X}{b:02X}"),
+            )
+            window.show()
+            # Add delay to ensure windows are visible after animation.
+            await main_window_probe.wait_for_window(
+                f"Test Window {i} is visible", full_screen=True
+            )
```

Comment:
There's no need for a pause after every window. Construct the windows; pause for the change to take effect once.

Reply:
Done, Thanks.